sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -347,8 +653,11 @@ class KernelProgramCache {
         BuildState NewState = BuildResult->waitUntilTransition();
 
         // Build succeeded.
-        if (NewState == BuildState::BS_Done)
+        if (NewState == BuildState::BS_Done) {
+          if constexpr (!std::is_same_v<EvictFT, void *>)
+            EvictFunc(BuildResult->Val, 0);
```

Comment:
Following my previous comment, I would prefer we do:
```suggestion
            EvictFunc(BuildResult->Val, /*IsBuilt=*/true);
```
Someone will have to read this in the future, so figuring out what the argument means without having to back-track is good.